broker/src/test/java/io/camunda/zeebe/broker/transport/externalapi/QueryApiRequestHandlerTest.java

```diff
@@ -0,0 +1,211 @@
+  @Test
+  void processNotFound() {
+    // given
+    final QueryApiRequestHandler sut = createQueryApiRequestHandler(true);
+    sut.addPartition(1, new QueryServiceReturning(null));
+
+    // when
+    final Either<ErrorResponse, ExecuteQueryResponse> response =
+        new AsyncExecuteQueryRequestSender(sut)
+            .sendRequest(
+                new ExecuteQueryRequest().partitionId(1).key(1).valueType(ValueType.PROCESS))
+            .join();
+
+    // then
+    EitherAssert.assertThat(response)
+        .isLeft()
+        .extracting(Either::getLeft)
+        .extracting(ErrorResponse::getErrorCode, ErrorResponse::getErrorData)
+        .containsExactly(
+            ErrorCode.PROCESS_NOT_FOUND,
+            "Expected to handle ExecuteQueryRequest, but no process found with key 1");
+  }
+
+  @DisplayName("should respond with bpmnProcessId when process found")
+  @Test
+  void processFound() {
+    // given
+    final QueryApiRequestHandler sut = createQueryApiRequestHandler(true);
+    sut.addPartition(1, new QueryServiceReturning("OneProcessToFindThem"));
+
+    // when
+    final Either<ErrorResponse, ExecuteQueryResponse> response =
+        new AsyncExecuteQueryRequestSender(sut)
+            .sendRequest(
+                new ExecuteQueryRequest().partitionId(1).key(1).valueType(ValueType.PROCESS))
+            .join();
+
+    // then
+    EitherAssert.assertThat(response)
+        .isRight()
+        .extracting(Either::get)
+        .extracting(ExecuteQueryResponse::getBpmnProcessId)
+        .isEqualTo("OneProcessToFindThem");
+  }
+
+  private static QueryApiRequestHandler createQueryApiRequestHandler(final boolean enabled) {
+    final var config = new QueryApiCfg();
+    config.setEnabled(enabled);
+    return new QueryApiRequestHandler(config);
+  }
+
+  private static final class AsyncExecuteQueryRequestSender {
+
+    private final QueryApiRequestHandler sut;
+    private final MessageHeaderDecoder messageHeaderDecoder = new MessageHeaderDecoder();
+    private final ExecuteQueryResponseDecoder queryResponseDecoder =
+        new ExecuteQueryResponseDecoder();
+
+    private int requestCount = 0;
+
+    public AsyncExecuteQueryRequestSender(final QueryApiRequestHandler requestHandler) {
+      sut = requestHandler;
+    }
+
+    private CompletableFuture<Either<ErrorResponse, ExecuteQueryResponse>> sendRequest(
+        final ExecuteQueryRequest queryRequest) {
+      final var future = new CompletableFuture<Either<ErrorResponse, ExecuteQueryResponse>>();
+      final ServerOutput serverOutput =
+          serverResponse -> {
+            // we need to write the serverResponse into a buffer in order to read it as SBE
+            final var bytes = new byte[serverResponse.getLength()];
+            final var buffer = new UnsafeBuffer(bytes);
+            serverResponse.write(buffer, 0);
+
+            messageHeaderDecoder.wrap(buffer, 0);
```

Comment:
:wrench: Reuse the `ExecuteQueryResponse` test utility for wrapping it to simplify